src/Libraries/CoreNodes/List.cs

```diff
@@ -525,6 +525,162 @@ public static IDictionary GroupByKey(IList list, IList keys)
             };
         }
 
+        /// <summary>
+        ///     This node clusters elements in a list based on the adjacency of their indices and the similarity of their values.
+        ///     It can handle list of numbers as well as list of strings.
+        ///     The boolean input allows the user to specify whether adjacency should be considered when grouping the elements.
+        ///     The tolerance input allows the user to specify the threshold value for grouping similar elements. In case of integers, the tolerance value is the difference between the two integers.
+        ///     And in case of strings, the tolerance value is the number of characters that can be different between the two strings.
+        /// </summary>
+        /// <param name="list">List of items to group as sublists based on adjacency and similarity</param>
+        /// <param name="tolerance">Threshold value for grouping similar items</param>
+        /// <param name="considerAdjacency">Boolean value to control if the node should consider adjacency of not.</param>
+        /// <returns name="groupedValues">list of sublists, with items grouped by same adjacent values</returns>
+        /// <returns name="groupedIndices">list of sublists, with items grouped by same adjacent indices</returns>
+        /// <search>list;group;similar;adjacent;adjacency;groupbyadjacency;groupadjacentitems;groupsimilaritems;cluster;tolerance</search>
+        [MultiReturn(new[] { "groupedValues", "groupedIndices" })]
+        [IsVisibleInDynamoLibrary(true)]
+        public static IDictionary GroupBySimilarity(IList list, int tolerance = 0, bool considerAdjacency = true)
+        {
+            // Validate input
+            if (list == null || list.Count == 0)
+                throw new ArgumentException("Need at least one item in the list of items.", nameof(list));
```

Comment:
I think we can return an empty list for an empty list as input instead of throwing an exception.